src/common/expression/Expression.cpp

```diff
@@ -741,4 +741,40 @@ std::ostream& operator<<(std::ostream& os, Expression::Kind kind) {
   return os;
 }
 
+Status PropertyTracker::update(const std::string& oldName, const std::string& newName) {
+  if (oldName == newName) {
+    return Status::OK();
+  }
+
+  auto it1 = vertexPropsMap.find(oldName);
+  bool has1 = it1 != vertexPropsMap.end();
+  auto it2 = edgePropsMap.find(oldName);
+  bool has2 = it2 != edgePropsMap.end();
+  if (has1 && has2) {
+    return Status::Error("Duplicated property name: %s", oldName.c_str());
+  }
```

Comment:
Don't introduce OptimizerError.

Reply:
The error just means the property pruner rule could not be done, and the error will not return to the user.